dotenv/godotenv.go

```diff
@@ -96,24 +96,27 @@ func load(overload bool, filenames ...string) (err error) {
 
 // ReadWithLookup gets all env vars from the files and/or lookup function and return values as
 // a map rather than automatically writing values into env
-func ReadWithLookup(lookupFn LookupFn, filenames ...string) (envMap map[string]string, err error) {
+func ReadWithLookup(lookupFn LookupFn, filenames ...string) (map[string]string, error) {
 	filenames = filenamesOrDefault(filenames)
-	envMap = make(map[string]string)
+	envMap := make(map[string]string)
 
 	for _, filename := range filenames {
 		individualEnvMap, individualErr := readFile(filename, lookupFn)
 
 		if individualErr != nil {
-			err = individualErr
-			return // return early on a spazout
+			return envMap, individualErr
 		}
 
 		for key, value := range individualEnvMap {
+			r := regexp.MustCompile(`^\s*\d.*`) // Keys starting with numbers are ignored
```

Comment:
Maybe move this to a module level variable or at least out of the loop?

Reply:
Done